src/crewai/crew.py

```diff
@@ -707,6 +724,75 @@ def _run_hierarchical_process(self) -> CrewOutput:
             token_usage=token_usage,
         )
 
+    def _process_async_tasks(
+        self,
+        futures: List[Tuple[Task, Future[TaskOutput], int]],
+        was_replayed: bool = False,
+    ) -> List[TaskOutput]:
+        task_outputs = []
+        for future_task, future, task_index in futures:
+            task_output = future.result()
+            task_outputs.append(task_output)
+            self._process_task_result(future_task, task_output)
+            self._store_execution_log(
+                future_task, task_output, task_index, was_replayed
+            )
+        return task_outputs
+
+    def _find_task_index(
+        self, task_id: str, stored_outputs: List[Any]
+    ) -> Optional[int]:
+        return next(
+            (
+                index
+                for (index, d) in enumerate(stored_outputs)
+                if d["task_id"] == str(task_id)
+            ),
+            None,
+        )
+
+    def replay_from_task(
```

Comment:
Does replay work if you run multiple crews consecutively? The reason I ask is because we are currently loading all stored outputs which can include task from three runs ago.

Reply:
We should probably add a few tests around the scenario I just described above.

Reply:
Nope. this will reset every kickoff. So we would need to do something else for replaying from a specific crew. that would be a different ticket tho